.azure-ci/build_manylinux2010.sh

```diff
@@ -1,3 +1,6 @@
 #!/bin/bash
-docker run -d -i -t -e python_ver=$PYTHON_VER --name manylinux10 -v `pwd`:/io quay.io/pypa/manylinux2010_x86_64 /bin/bash
-docker exec manylinux10 sh -c "sh /io/.azure-ci/docker_scripts.sh"
+docker run -t --rm -e python_ver=$PYTHON_VER \
+	-v `pwd`:/io \
+	-v "${CCACHE_DIR}":/root/.ccache/  \
+	quay.io/pypa/manylinux2010_x86_64 \
+	/bin/bash -c "bash /io/.azure-ci/docker_scripts.sh"
```

Comment:
Main motivation of this change is to facilitate running this script locally. The previously version kept a container named `manylinux10` in the background that needed to be manually killed, and re-running the script produced an error as the name was already taken.

This changes automatically deletes the container after the command has run.